contrib/containerized-pbench/README.md

```diff
@@ -1,81 +1,118 @@
+$ oc patch daemonset pbench-agent --patch \ '{"spec":{"template":{"spec":{"serviceAccountName": "useroot"}}}}'
 ```
 
-By default pod is not scheduled on the master node since the scheduling is disabled. Since we are using daemonset, pods created by the Daemon controller have the machine already selected (.spec.nodeName is specified when the pod is created, so it is ignored by the scheduler) which means that the unschedulable field of a node is not respected by the DaemonSet controller and DaemonSet controller can make pods even when the scheduler has not been started. For this to work we have to create a service account and add it to the privileged scc like
+## Prepare the jump host to run pbench-controller
 
+### Get the controller image from dockerhub
 ```
-$ oc create serviceaccount useroot
-$ oc adm policy add-scc-to-user privileged -z useroot
-$ oc patch daemonset pbench-agent --patch '{"spec":{"template":{"spec":{"serviceAccountName": "useroot"}}}}'
+$ docker pull ravielluri/image:controller
+$ docker tag ravielluri/image:controller pbench-controller:latest
 ```
 
-For pbench pod to run on the lb, we have configure it as a node ( node+lb ), disable scheduling on lb:
-
+### keys
+copy the ssh keys to /root/scale-testing/keys. The keys directory should contain a perf key named as id_rsa_perf,  id_rsa - the private key needed to copy the results to the pbench server, authorized_keys file -ansible needs to have a passwordless authentication to localhost inside the container, ssh config which looks like:
 ```
-$ oadm manage-node node1.example.com --schedulable=false
+Host *
+	User root
+        Port 2022
+        StrictHostKeyChecking no
+        PasswordAuthentication no
+        UserKnownHostsFile ~/.ssh/known_hosts
+        IdentityFile ~/.ssh/id_rsa_perf  
+      
+Host *pbench-server
+        User root
+        Port 22
+        StrictHostKeyChecking no
+        PasswordAuthentication no
+        UserKnownHostsFile ~/.ssh/known_hosts
+        IdentityFile /opt/pbench-agent/id_rsa
 ```
 
-Also make sure you label your nodes with type:pbench so that daemonset knows where to schedule the pbench-pod:
-
-```
-$ oc label node <node> type=pbench
+### Inventory
+Make sure you have the inventory used to install openshift and it should look like:
 ```
+[pbench-controller]
 
-## Jump node
+[masters]
+    
+[nodes]
 
-### Build pbench-controller image 
+[etcd]
 
-```
-$ cd containerized-pbench/dockerfiles-jump_node 
-$ docker build -t pbench-controller .
+[lb]
+
+[prometheus-metrics]
+<host> port=8443 cert=/etc/origin/master/admin.crt key=/etc/origin/master/admin.key
+<host> port=10250 cert=/etc/origin/master/admin.crt key=/etc/origin/master/admin.key
+
+[pbench-controller:vars]
+register_all_nodes=False
 ```
 
-### Run the container
+Set register_all_nodes to true if the tools needs to be registered on all the nodes, if not set to true, it registers pbench tools on just two of the nodes.
```

Comment:
Are "two" and "all" the only possible choices?

Reply:
Yes they are the only possible choices.